[blog] object-calisthenics init

TomasVotruba opened the pull request:
Post about release: https://github.com/object-calisthenics/phpcs-calisthenics-rules/releases/tag/v3.0.0


## Todo

- [x] redate to 2017-06-26

TomasVotruba commented:
@UFOMelkor I've prepared a small post about Object Calisthenics 3.0 and as you've helped me with this release, I'd like to ask you for a review. Woudl you add anything, remove anything etc.

Thanks :heart:

TomasVotruba commented:
@frenck @mihaeu @roukmoute @UFOMelkor
I've prepared a small post about Object Calisthenics 3.0 and as you've helped me with this release, I'd like to ask you for a review. Woudl you add anything, remove anything etc.

Thanks :heart:

mihaeu commented:
Hey, thanks. I'll have a look at it tonight.

Cheers,
Michael

On Jun 18, 2017 11:53, "Tomáš Votruba" <notifications@github.com> wrote:

> @frenck <https://github.com/frenck> @mihaeu <https://github.com/mihaeu>
> @roukmoute <https://github.com/roukmoute> @UFOMelkor
> <https://github.com/ufomelkor>
> I've prepared a small post about Object Calisthenics 3.0 and as you've
> helped me with this release, I'd like to ask you for a review. Woudl you
> add anything, remove anything etc.
>
> Thanks ❤️
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/TomasVotruba/tomasvotruba.cz/pull/110#issuecomment-309267487>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACEXfWJIfhzKCdUYVwgTa9MES-pkvAE0ks5sFPOtgaJpZM4N9Tfx>
> .
>

TomasVotruba commented:
@mihaeu Hey Michael, do you think you can make it by the end of the week?
I'd like to publish it for mondays.

mihaeu commented:
Tonight's definitely the night 😉

On Jun 21, 2017 14:19, "Tomáš Votruba" <notifications@github.com> wrote:

> @mihaeu <https://github.com/mihaeu> Hey Michael, do you think you can
> make it by the end of the week?
> I'd like to publish it for mondays.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/TomasVotruba/tomasvotruba.cz/pull/110#issuecomment-310061252>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACEXfUVbETSBna1T4OWuRabyeF27-l8Oks5sGQpngaJpZM4N9Tfx>
> .
>

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+layout: post
+title: "PHP Object Calisthenics Made Simple - Version 3.0 is Out Now"
+perex: '''
+    Object Calisthenics are 9 language-agnostic rules to help you write better and cleaner code.
```
technically you'd have to spell out numbers less than 12 ;)

TomasVotruba replied:
That's rather matter of copywriting :). Numbers are easier to scan.

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+title: "PHP Object Calisthenics Made Simple - Version 3.0 is Out Now"
+perex: '''
+    Object Calisthenics are 9 language-agnostic rules to help you write better and cleaner code.
+    They help you to get rid of else, method chaining, long classes or functions, unreadable short names and so on.
```
- else-statements
 - and much more.

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+    Object Calisthenics are 9 language-agnostic rules to help you write better and cleaner code.
+    They help you to get rid of else, method chaining, long classes or functions, unreadable short names and so on.
+    <br><br>
+    <strong>Object Calisthenics 3.0 runs on CodeSniffer 3.0, PHP 7.1 and brings 6 of them with fancy configuration and code examples</strong>.
```
- bring is the wrong verb here, maybe ship?
 -  break it up in two sentences. the second one about configuration and code examples

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+---
+layout: post
+title: "PHP Object Calisthenics Made Simple - Version 3.0 is Out Now"
+perex: '''
```
do you already have a post about the rules? maybe link to some sources?

TomasVotruba replied:
Not yet, this is the first post here.
There is a link in first paragraph to external source.

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+lang: en
+---
+
+If you are coding standard nerd like me, you'll probably more than PSR2 standards in your ruleset. But even if you fon't, [Object Calisthenics](https://github.com/object-calisthenics/phpcs-calisthenics-rules) is developer-friendly game changer for your code.
```
- you are **a** coding standard nerd
 - probably **have** more ... than **just PSR-2** standards
 - fon't -> don't
- ... is **a** developer-friendly

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+
+## Much Simpler than 2.0
+
+**1. You don't have to know single thing about CodeSniffer to start**
```
**a** single thing

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+
+**[Rule #6 - Do Not Abbreviate](https://github.com/object-calisthenics/phpcs-calisthenics-rules#6-do-not-abbreviate)** checks these cases. **It detects short names that are ambiguous and hard to decode.**
+
+To run it in CLI just include Object Calisthenics' `ruleset.xml` and specify the sniff name: 
```
from the command line ... the sniff's name

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
````diff
@@ -0,0 +1,155 @@
+--sniffs=ObjectCalisthenics.NamingConventions.ElementNameMinimalLength
+```
+
+You can run this locally or put to your CI integration and you are ready to go.
````
put it in your CI configuration (the I in CI already stands for integration)

TomasVotruba replied:
Good point :

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+
+## Fancy Readme With Examples
+
+We put lot of work to README for new release. It isn't a long text describing what exactly the rule does and how it originated - there is already [blog post for that](http://williamdurand.fr/2013/06/03/object-calisthenics/). 
```
- lots of work in the README ... the new release
 - is already **a**

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+- [@UFOMelkor](https://github.com/UFOMelkor)
+
+Without your help, this would not have been possible.
+Thank you guys.
```
you're welcome ;)

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+
+## To Sum up Object Calisthenics 3.0
+
+- This release focuses **on simple setup** rather than on academic explanation of their meaning.  
```
if you want to make it shorter and more memorable consider phrasing it like the agile manifesto:

simple setup over boring academic explanations

TomasVotruba replied:
Love it!

mihaeu commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -0,0 +1,155 @@
+
+- This release focuses **on simple setup** rather than on academic explanation of their meaning.  
+- Rules are *explained in examples*.
+- 0-setup. You can just **copy-paste CLI command and run it**.
```
either: 
 - a cli command
or
 - cli commands

mihaeu commented:
Done. Made a couple of remarks.

TomasVotruba commented:
@mihaeu Thanks for extensive feedback, I'm on it!

TomasVotruba commented:
> Content is fine, but maybe a bit more about the rules themselves if you haven't already posted about them. I know you don't want to repeat yourself, but a reader who's never heard about object calisthenics before might not be motivated to read up about it on another site.

I understand what you mean: more content to draw in first-time user.

What exactly do you suggest? Addin 2 rules? If so, which ones do you suggest?
I'm too deep in the development so I don't realize what rules are "the best" to promote.

TomasVotruba commented:
@mihaeu Ping?

mihaeu commented:
Hey Tomas, I got overwhelmed by too many notifications and then just ended up deleting them :)

What you could also do is just list them instead. But my personals favorites are "no setters" and "wrap all primitives"

Let me know if you want a final review tomorrow.

Cheers,
Michael

TomasVotruba commented:
> Hey Tomas, I got overwhelmed by too many notifications and then just ended up deleting them :)

I know what you mean. I only use Github notifications in the top right, no email spam anymore :)

> What you could also do is just list them instead. But my personals favorites are "no setters" and "wrap all primitives"

I've added them with with NO/YES examples and link to readme for more configuration.

I've also added 3 missing rules, as 9 - 6 might confuse somebody.

I would rather not add any more text, as the post is already quite long for first-timer.

Let me know :+1: It's ready for merge for me.

roukmoute commented on source/_posts/2017/2017-06-17-php-object-calisthenics-rules-made-simple-version-3-0-is-out-now.md:
```diff
@@ -130,6 +130,213 @@ checkers:
+    return true;
+}
+
+return false;
```
I don't think it's a good example.
Because here just do `return $isEnabled`

TomasVotruba replied:
What would you suggest?

roukmoute replied:
In the book they give two example.
A ternary like here finally or two different actions.

Anything like that?

```php
if ($status == self::DONE) {
    $this->finish();
    return;
}
$this->advance($step);
```

TomasVotruba replied:
I see, I think it makes sense. Could you send PR here and to the README of the package?

Just with strict comparison `===` to make it more fail proof

roukmoute replied:
Yep of course :)

TomasVotruba replied:
That's great :+1:  I'll get to it in the evening. 

I can merge it and publish it.

mihaeu commented:
I didn't follow the whole conversation but why do you use an if statement
in that example? return $isEnabled; is all you need

On Jun 27, 2017 14:24, "Mathias STRASSER" <notifications@github.com> wrote:

> *@roukmoute* commented on this pull request.
> ------------------------------
>
> In source/_posts/2017/2017-06-17-php-object-calisthenics-rules-
> made-simple-version-3-0-is-out-now.md
> <https://github.com/TomasVotruba/tomasvotruba.cz/pull/110#discussion_r124257361>
> :
>
> > +```php
> +if ($isEnabled) {
> +    return true;
> +} else {
> +    return false;
> +}
> +```
> +
> +<em class="fa fa-fw fa-check"></em>
> +
> +```php
> +if ($isEnabled) {
> +    return true;
> +}
> +
> +return false;
>
> Quick!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/TomasVotruba/tomasvotruba.cz/pull/110#discussion_r124257361>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACEXfT2d4lx-un5n87s7XttlJIjEXfugks5sIPRwgaJpZM4N9Tfx>
> .
>

roukmoute commented:
@mihaeu follow the whole conversation :wink:

roukmoute commented:
God I've completly forgot to add `===` in my P.R. :expressionless:

Could you merge our two same commits and add missing equal?

TomasVotruba commented:
That happens :angel: . `===` added. I will squash this PR to master anyway.

Could you please also add it to README? Ad

TomasVotruba commented:
Merged! Thank you both for feedback and your work here.
I feel good about this teamwork :parasol_on_ground: 

Feel free to send PR, if you'll find any more mistakes.

TomasVotruba commented:
Post is here: https://www.tomasvotruba.cz/blog/2017/06/26/php-object-calisthenics-rules-made-simple-version-3-0-is-out-now/

Tweet or share is appreciated :+1:
